### PR TITLE
sonarscan reusable workflow added

### PR DESCRIPTION
**Pull request by ng-htec**
readme added for sonarscan

### CODE REVIEW

File: sonarscan/sonar-scan-dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        required: true
+      GITHUB_TOKEN:
+        required: true
+jobs:
+
+  sonarcloud:
+    name: Code Quality Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (jz-htec) [inline sonarscan/sonar-scan-dotnet.yml:34]: GitHub token should be also utilized in checkout phase

File: sonarscan/sonar-scan-dotnet.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        required: true
+        type: string
+        default: '11'
+      PROJECT_NAME:
+        required: true
+        type: string 
+      ORG_NAME:
+        required: true
+        type: string
+      BUILD_DIR:
```
Comment (jz-htec) [inline sonarscan/sonar-scan-dotnet.yml:20]: Default value should be project root directory

**Review by jz-htec — CHANGES_REQUESTED** [review]

File: utils/action.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: 'Repository data'
+description: 'Fetch repository and organization name'
+
+outputs:
+  repo_name:
```
Comment (jz-htec) [inline utils/action.yml:5]: Consider also to export other available variables, such as APP_VERSION, or APP_REVISION, etc.

File: sonarscan/dotnet.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        type: string
+        default: "./"
+      repo_name:
+        required: true
+        type: string 
+      org_name:
+        required: true
+        type: string
+    secrets:
+      SONAR_TOKEN:
```
Comment (jz-htec) [inline sonarscan/dotnet.yml:22]: I think it would be better to use lowercase secrets name to avoid confusion with ENVs

**Review by jz-htec — COMMENTED** [review]
Few more comments on the PR just to polish everything

File: sonarscan/dotnet.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+name: Sonar scan
+
+on:
+  workflow_call:
+    inputs:
+      DOTNET_VERSION:
```
[the remark this answers is not in the document]
Comment (ng-htec) [reply sonarscan/dotnet.yml]: I have resolved that on previous commit https://github.com/htec-infra/gha-tbd/commit/53a1bb6abc72767b7d1205021b49b684230be992#diff-19e398a45e0b6d682d7c64cc29a699341d0f133fbbe711c5a5b78b72e812e775

File: sonarscan/dotnet.yml (hunk 4 above)
Comment (ng-htec) [reply sonarscan/dotnet.yml:22]: Agree with you. It will be fixed on next push

File: utils/action.yml (hunk 3 above)
Comment (ng-htec) [reply utils/action.yml:5]: Will be added on next push

**Review by jz-htec — APPROVED** [review]
Excellent!

**Pull request merged**